src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -405,6 +405,50 @@ async Task IDurableOrchestrationClient.TerminateAsync(string instanceId, string
             }
         }
 
+        async Task IDurableOrchestrationClient.SuspendAsync(string instanceId, string reason)
+        {
+            OrchestrationState state = await this.GetOrchestrationInstanceStateAsync(instanceId);
+            if (IsOrchestrationSuspendable(state))
+            {
+                state.OrchestrationInstance.ExecutionId = null;
+
+                await this.client.SuspendInstanceAsync(state.OrchestrationInstance, reason);
+
+                this.traceHelper.FunctionSuspended(this.TaskHubName, state.Name, instanceId, reason);
```

Comment:
"FunctionSuspended" is a little misleading since we don't know whether the orchestration has actually gone into the suspended state at this point. Saying that the "function" is suspended is also a little strange because we don't suspend "functions" - rather, we suspend orchestrations. A better log message would be "SuspendingOrchestration". I also suggest putting this log statement before the call to send the suspend message.

Reply:
Got it. I just tried to follow the naming pattern of "FunctionTerminated". I am assuming there is a difference there, then, that I didn't realize? Anyway, I have changed the name now.

Reply:
Ah, shoot, sorry for the inconsistency. I still think it's best to use `ing` instead of `ed`, so let's do that (as you have already done) for these new APIs.